zerver/tests/test_digest.py

```diff
@@ -540,6 +557,48 @@ def test_new_stream_link(self) -> None:
         self.assertEqual(stream_count, 0)
         self.assertEqual(stream_info["html"], [])
 
+    def test_get_new_messages_count(self) -> None:
+        othello = self.example_user("othello")
+        self.subscribe(othello, "Verona")
+
+        cutoff = timezone_now() - timedelta(days=5)
+        old_human_messages_count = get_new_messages_count(othello, cutoff)
+
+        senders = ["hamlet", "cordelia", "iago", "prospero", "ZOE", "default_bot"]
+        # Exclude messages sent by bots from digests, as only human messages should be considered.
+        new_human_messages_count = len(self.simulate_stream_conversation("Verona", senders)) - 1
+
+        self.assertEqual(
+            get_new_messages_count(othello, cutoff),
+            old_human_messages_count + new_human_messages_count,
+        )
+
+    @mock.patch("zerver.lib.digest.send_future_email")
+    def test_email_digest_when_message_content_is_disabled(
+        self, mock_send_future_email: mock.MagicMock
+    ) -> None:
+        realm = get_realm("zulip")
+        othello = self.example_user("othello")
+        cutoff_date = timezone_now() - timedelta(days=5)
+        cutoff = time.mktime(cutoff_date.timetuple())
+        do_set_realm_property(
+            realm, "message_content_allowed_in_email_notifications", False, acting_user=None
+        )
+
+        # When this test is run in isolation, one additional query is run which
+        # is equivalent to
+        # ContentType.objects.get(app_label='zerver', model='userprofile')
+        # This code is run when we call `confirmation.models.create_confirmation_link`.
+        # To trigger this, we call the one_click_unsubscribe_link function below.
+        one_click_unsubscribe_link(othello, "digest")
+
+        # Clear the LRU cache on the stream topics
+        get_recent_topics.cache_clear()
+        with self.assert_database_query_count(8):
+            bulk_handle_digest_email([othello.id], cutoff)
+
+        self.assertEqual(mock_send_future_email.call_count, 1)
```

Comment:
Maybe check a little bit more of the context that's passed for this mocked future email. Like the parameter for disabling / hiding message content.